### PR TITLE
Fast path the merge querier 

### LINKED ISSUE #3065: Query slow down in v2 due to remote read
Reported by @fabxc on IRC: 

```
tomwilkie[m]: the remote storage bits are giving a notable query overhead even if no remote storages are configured =/
```

### PR DESCRIPTION
**Pull request by tomwilkie**
Such that it is completely removed from query path when there is no remote storage.

See #3357, actually fix #3065.

### CODE REVIEW

**Comment by tomwilkie** [comment]
I don't feel like returning nil is a very friendly interface to consume, so I would consider returning a sentinel `noopQuerier` and filtering that out instead, if anyone feels strongly.

File: storage/fanout.go [hunk 1]
```diff
@@ -197,9 +197,26 @@ type mergeQuerier struct {
 }
 
 // NewMergeQuerier returns a new Querier that merges results of input queriers.
+// NB will return nil if no queriers are passed to it, and will filter nil
```
Comment (fabxc) [inline storage/fanout.go]: NewMergeQuerier

**Comment by tomwilkie** [comment]
> noopQuerier would be cleaner and probably be useful for future things as well. But not feeling strongly for now.

I added it.

File: storage/noop.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// limitations under the License.
+
+package storage
+
+import "github.com/prometheus/prometheus/pkg/labels"
+
+type noopQuerier struct{}
+
+// NoopQuerier is a Querier that does nothing.
+var NoopQuerier Querier = noopQuerier{}
```
Comment (fabxc) [inline storage/noop.go]: To avoid further global variables, how about making this a function?

**Comment by fabxc** [comment]
👍  thanks

**Pull request merged**